File: src/main/java/no/nav/syfo/mock/AktoerMock.java
package no.nav.syfo.mock;

import no.nav.tjeneste.virksomhet.aktoer.v2.AktoerV2;
import no.nav.tjeneste.virksomhet.aktoer.v2.HentAktoerIdForIdentPersonIkkeFunnet;
import no.nav.tjeneste.virksomhet.aktoer.v2.HentIdentForAktoerIdPersonIkkeFunnet;
import no.nav.tjeneste.virksomhet.aktoer.v2.meldinger.*;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Service;

import static no.nav.syfo.config.AktoerConfig.MOCK_KEY;

@Service
@ConditionalOnProperty(value = MOCK_KEY, havingValue = "true")
public class AktoerMock implements AktoerV2 {

    private static final String MOCK_AKTORID_PREFIX = "10";

    public WSHentAktoerIdForIdentResponse hentAktoerIdForIdent(WSHentAktoerIdForIdentRequest request) throws HentAktoerIdForIdentPersonIkkeFunnet {
        return new WSHentAktoerIdForIdentResponse()
                .withAktoerId(mockAktorId(request.getIdent()));
    }

    public WSHentIdentForAktoerIdResponse hentIdentForAktoerId(WSHentIdentForAktoerIdRequest request) throws HentIdentForAktoerIdPersonIkkeFunnet {
        return new WSHentIdentForAktoerIdResponse().withIdent(getFnrFromMockedAktorId(request.getAktoerId()));
    }

    public WSHentAktoerIdForIdentListeResponse hentAktoerIdForIdentListe(WSHentAktoerIdForIdentListeRequest request) {
        throw new RuntimeException("Denne er ikke implementert i mocken");
    }

    public WSHentIdentForAktoerIdListeResponse hentIdentForAktoerIdListe(WSHentIdentForAktoerIdListeRequest request) {
        throw new RuntimeException("Denne er ikke implementert i mocken");
    }

    public void ping() {
    }

    public static String mockAktorId(String fnr) {
        return MOCK_AKTORID_PREFIX.concat(fnr);
    }

    private static String getFnrFromMockedAktorId(String aktorId) {
        return aktorId.replace(MOCK_AKTORID_PREFIX, "");
    }
}

File: src/main/kotlin/no/nav/syfo/consumer/aktorregister/AktoerMock.kt
package no.nav.syfo.consumer.aktorregister

import no.nav.tjeneste.virksomhet.aktoer.v2.*
import no.nav.tjeneste.virksomhet.aktoer.v2.meldinger.*
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Service

@Service
@ConditionalOnProperty(value = [AktorConfig.MOCK_KEY], havingValue = "true")
class AktoerMock : AktoerV2 {
    @Throws(HentAktoerIdForIdentPersonIkkeFunnet::class)
    override fun hentAktoerIdForIdent(request: WSHentAktoerIdForIdentRequest): WSHentAktoerIdForIdentResponse {
        return WSHentAktoerIdForIdentResponse()
            .withAktoerId(mockAktorId(request.ident))
    }

    @Throws(HentIdentForAktoerIdPersonIkkeFunnet::class)
    override fun hentIdentForAktoerId(request: WSHentIdentForAktoerIdRequest): WSHentIdentForAktoerIdResponse {
        return WSHentIdentForAktoerIdResponse().withIdent(getFnrFromMockedAktorId(request.aktoerId))
    }

    override fun hentAktoerIdForIdentListe(request: WSHentAktoerIdForIdentListeRequest): WSHentAktoerIdForIdentListeResponse {
        throw RuntimeException("Denne er ikke implementert i mocken")
    }

    override fun hentIdentForAktoerIdListe(request: WSHentIdentForAktoerIdListeRequest): WSHentIdentForAktoerIdListeResponse {
        throw RuntimeException("Denne er ikke implementert i mocken")
    }

    override fun ping() {}

    companion object {
        private const val MOCK_AKTORID_PREFIX = "10"
        @JvmStatic
        fun mockAktorId(fnr: String): String {
            return MOCK_AKTORID_PREFIX + fnr
        }

        private fun getFnrFromMockedAktorId(aktorId: String): String {
            return aktorId.replace(MOCK_AKTORID_PREFIX, "")
        }
    }
}
